magefile.go

```diff
@@ -961,27 +952,41 @@ func runAgent(env map[string]string) error {
 	return sh.Run("docker", dockerCmdArgs...)
 }
 
-var platformPackages = map[string]string{
-	"darwin/amd64":  "darwin-x86_64.tar.gz",
-	"darwin/arm64":  "darwin-aarch64.tar.gz",
-	"linux/amd64":   "linux-x86_64.tar.gz",
-	"linux/arm64":   "linux-arm64.tar.gz",
-	"windows/amd64": "windows-x86_64.zip",
-}
-
-func packageAgent(platforms []string, dependenciesVersion string, agentPackaging, agentBinaryTarget mg.Fn) {
+func packageAgent(ctx context.Context, platforms []string, dependenciesVersion string, agentPackaging, agentBinaryTarget mg.Fn) {
 	fmt.Println("--- Package Elastic-Agent")
+	var manifestResponse tools.Build
+	var err error
 
 	requiredPackages := []string{}
 	for _, p := range platforms {
-		requiredPackages = append(requiredPackages, platformPackages[p])
+		requiredPackages = append(requiredPackages, manifest.PlatformPackages[p])
 	}
 	if mg.Verbose() {
 		log.Printf("--- Packaging dependenciesVersion[%s], %+v \n", dependenciesVersion, requiredPackages)
 	}
 
+	if devtools.PackagingFromManifest {
```

Comment:
I would move this block in `PackageUsingDRA` target (that's what we are gonna use for independent releases anyways and we already have to download the manifest ) and `Package` target  (in case someone want to still build the agent binary but use the dependencies from a manifest) and pass in a pointer to a `tools.Build` struct (possibly nil if no MANIFEST_URL is there)

Reply:
Done in https://github.com/elastic/elastic-agent/pull/4885/commits/7754ffdc96a0cc14e2148b0ab35e3419fdc770ba